apps/site/lib/site_web/controllers/schedule/line_api.ex

```diff
@@ -157,14 +174,15 @@ defmodule SiteWeb.ScheduleController.LineApi do
   # for each list of disrupted stops, make adjustment to the diagram based on
   # the nature of the disruption. e.g. for detour or stop/station closure we
   # modify the stop preceding it but for shuttle we don't style the final stop
+  @spec shift_indices_by_disruption_effect([number], [Alerts.Alert.effect()]) :: [number]
   defp shift_indices_by_disruption_effect(indices, effects) do
-    effect_avoids_stop? = effects -- effects -- [:stop_closure, :station_closure, :detour]
+    effects_avoiding_stop = effects -- effects -- [:stop_closure, :station_closure, :detour]
 
     cond do
       :shuttle in effects ->
         List.delete_at(indices, -1)
 
-      effect_avoids_stop? && List.first(indices) > 0 ->
+      Kernel.length(effects_avoiding_stop) > 0 && List.first(indices) > 0 ->
```

Comment:
This change addresses my previous comment. I think you can probably switch back to `and` if you want.